src/gpuhunt/providers/gcp.py

```diff
@@ -232,3 +262,266 @@ def filter(cls, offers: List[RawCatalogItem]) -> List[RawCatalogItem]:
+                    tpu_version, no_of_cores, location, no_of_chips, False
+                )
+            if spot_base_price is not None:
+                spot_price = spot_base_price * no_of_chips
+            else:
+                spot_price = find_tpu_price_static_src(
+                    tpu_version, no_of_cores, location, no_of_chips, True
+                )
+        elif any(tpu_version == x for x in ["v2", "v3", "v4"]):
+            # For TPU-v2 and TPU-v3, the pricing API provides the prices of 8 TPU cores.
+            # For TPU-v4, api only provides the price of TPU-v4 pods.
+            if no_of_cores > 8 or tpu_version == "v4":
+                is_pod = True
+                base_instance_name = f"{tpu_version}-8"
+                base_no_of_chips = find_no_of_chips(base_instance_name, configs)
+                on_demand_base_price = find_base_price(
+                    tpu_version, location, tpu_prices, spot=False, is_pod=True
+                )
+                spot_base_price = find_base_price(
+                    tpu_version, location, tpu_prices, spot=True, is_pod=True
+                )
+
+                if on_demand_base_price is not None and base_no_of_chips is not None:
+                    on_demand_price = (on_demand_base_price / base_no_of_chips) * no_of_chips
+                else:
+                    on_demand_price = find_tpu_price_static_src(
+                        tpu_version, no_of_cores, location, no_of_chips, False
+                    )
+                if spot_base_price is not None and base_no_of_chips is not None:
+                    spot_price = (spot_base_price / base_no_of_chips) * no_of_chips
+                else:
+                    spot_price = find_tpu_price_static_src(
+                        tpu_version, no_of_cores, location, no_of_chips, True
+                    )
+
+            elif no_of_cores == 8:
+                is_pod = False
+                on_demand_base_price = find_base_price(
+                    tpu_version, location, tpu_prices, spot=False, is_pod=False
+                )
+                spot_base_price = find_base_price(
+                    tpu_version, location, tpu_prices, spot=True, is_pod=False
+                )
+                on_demand_price = (
+                    on_demand_base_price if on_demand_base_price is not None else None
+                )
+                spot_price = spot_base_price if spot_base_price is not None else None
+                base_no_of_chips = no_of_chips
+
+        config["price"] = on_demand_price
+        config["spot"] = spot_price
+        config["is_pod"] = is_pod
+        config["base_price"] = on_demand_base_price
+        config["base_no_of_chips"] = base_no_of_chips
+        config["spot_base_price"] = spot_base_price
+    return configs
+
+
+def get_tpu_prices() -> List[dict]:
+    client = CloudCatalogClient()
+    tpu_configs = []
+
+    service_names = ["services/E000-3F24-B8AA", "services/6F81-5844-456A"]
```

Comment:
I see E000-3F24-B8AA is for TPU SKU group and 6F81-5844-456A is for On-Demand VMs group. What is the reason to list 6F81-5844-456A SKUs? Do they contain TPUs too?

Reply:
@r4victor E000-3F24-B8AA contains prices for TPU versions v2,v3,v4. Prices for TPU versions v5p and v5litepod are in 6F81-5844-456A

Reply:
Please leave a comment on this – this is valuable information.